### ISSUE TITLE
Reading

### ISSUE THREAD

**Issue by phamquiluan** [phamquiluan/FacialExpressionRecognition#1]

**Comment by phamquiluan** [comment]
https://arc.aiaa.org/doi/pdf/10.2514/6.2019-3411
Dự đoán năng suất làm việc của con người **real-time**, chưa có tải được paper nữa, hic hic.

http://ssba.org.se/ssdl2019/abstracts/Torstensson_etal.pdf
Nhận diện hành động của cha tài xế và cả đám hành khách

**Comment by phamquiluan** [comment]
Development of a facial‐expression database of Chinese Han, Hui and Tibetan people
(https://onlinelibrary.wiley.com/doi/abs/10.1002/ijop.12602), Lại là Prof Zhao @@ 

Tập dữ liệu được thu thập từ 200 người Hán, 220 người Hồi, 210 người Tạng. Ngoài ra còn có survey other works rất chi tiết

**Comment by phamquiluan** [comment]
MobileNets: Efficient Convolutional Neural Networks for Mobile Vision Applications
(https://arxiv.org/abs/1704.04861) (Submitted on 17 Apr 2017)

TODO: reading

**Comment by phamquiluan** [comment]
https://medium.com/datadriveninvestor/real-time-facial-expression-recognition-f860dacfeb6a

**Comment by phamquiluan** [comment]
A Deep Learning System for Recognizing Facial Expression in Real-Time
(https://dl.acm.org/citation.cfm?id=3311747)

Journal
--
ACM Transactions on Multimedia Computing, Communications, and Applications (TOMM) TOMM Homepage archive
Volume 15 Issue 2, June 2019 Article No. 33


Có center loss, real time, transfer learning

**Comment by phamquiluan** [comment]
Có một cái cần phải đọc để biết là cái khâu detection :))

**Comment by phamquiluan** [comment]
https://arxiv.org/abs/1906.06818
Stacked Capsule Autoencoders
Adam R. Kosiorek, Sara Sabour, Yee Whye Teh, Geoffrey E. Hinton

Một đối tượng có thể được xem là một tập hợp có tổ chức geometrically, một cái hệ thống robust có thể nhìn ra được điều này mà phân biệt, bởi vì những cái nội lực trong bản thân nó là bất biến. 

Đề xuất ra 1 bộ AE có tên là SCAE có 2 giai đoạn,

- Part Capture AE, phân đoạn ảnh thành các thành phần, suy luận góc độ và xây dựng lại các pixel ảnh như một hỗn hợp được chuyển đổi lại từ các part templaté.
- Object Capsule AE, cố gắng tổ chức các phần được khai thác (discovered parts) và góc độ của nó (their poses) thành các thành phần con mà có thể giải thích được và predict trên cái đám đó, rồi sau đó kết hợp cái đám predict này lại với nhau để ra được kết quả cuối cùng.

SCAE bắt được các mối quan hệ spatial của toàn bộ đối tượng và các thành phần của nó khi được huấn luyện trên dữ liệu không nhãn. Các véc-tơ thể hiện đấy sẽ được gom cụm hay làm cái gì đấy, mà có sure là nó separate không ta ???

**Comment by phamquiluan** [comment]
https://arxiv.org/pdf/1710.05941.pdf
Searching for Activation Functions
Prajit Ramachandran, Barret Zoph, Quoc V. Le
(Submitted on 16 Oct 2017 (v1), last revised 27 Oct 2017 (this version, v2))

Đề xuất một kỹ thuật tìm kiếm activation function sử dụng kết hợp các giải thuật học tham lam và học tăng cường (exhautive and reinforcement)

**Comment by phamquiluan** [comment]
Softmax: 


$$a_i$$

**Comment by phamquiluan** [comment]
ROC Curve là gì? - Đường cong đặc trưng hoạt động của bộ phận thu nhận. Trong lĩnh vực tín hiệu nó được dùng để phát hiện có tín hiệu hay chỉ là nhiễu.

![image](https://user-images.githubusercontent.com/24642166/60314357-8fe2fe00-998c-11e9-86b9-8e86be9a08bd.png)


Đồ thị có 2 trục: Đặc trưng (Specificity) và Độ nhạy (Sensitivity), sẽ có một cái ngưỡng cứng để xác định bên nhiễu và bên tín hiệu. Vậy cách đặt ngưỡng như thế nào? Và định nghĩa đặc trưng là độ nhạy là gì?

note1: thay đổi ngưỡng sẽ thay đổi true positives, true negatives, false positives, false negatives 


(http://gim.unmc.edu/dxtests/roc2.htm)
(https://towardsdatascience.com/understanding-auc-roc-curve-68b2303cc9c5)
(https://vi.wikipedia.org/wiki/%C4%90%C6%B0%E1%BB%9Dng_cong_ROC)
(https://en.wikipedia.org/wiki/Receiver_operating_characteristic)

**Comment by phamquiluan** [comment]
Python memory managements: https://rushter.com/blog/python-memory-managment/

Python sử dụng một bộ cấp phát có tên PyMalloc, minh họa mô hình phân cấp như bên dưới.

![image](https://user-images.githubusercontent.com/24642166/60392255-20564580-9b29-11e9-8f12-535b9c0846c0.png)

**Comment by phamquiluan** [comment]
Khái niệm **overhead** trong computer science, overhead is any combination of excess or indirect computation time, memory, bandwidth or other resources. It is a special case of engineering overhead.

**Comment by phamquiluan** [comment]
### Docker
https://vn.apps-gcp.com/docker-la-gi-tai-sao-phai-dung-docker/
https://viblo.asia/p/tim-hieu-ve-docker-cho-nguoi-moi-bat-dau-yMnKMAnAK7P (https://runnable.com/docker/, https://docs.docker.com/, https://viblo.asia)
https://ithot.vn/threads/docker-la-gi-vi-sao-va-khi-nao-dung-docker.46/
https://blog.duyet.net/2015/12/docker-la-gi-co-ban-ve-docker.html#.XRnAdugzZdg
https://minhng.info/docker/docker-la-gi.html
https://techtalk.vn/blog/posts/docker-la-gi
https://jobs.evolable.asia/blog/docker-la-gi/


Questions??
1. Tại sao docker lại không bị phụ thuộc os, nó vẫn phải tương tác với os mà ? 


**Build once, run anywhere**

**Comment by phamquiluan** [comment]
### Non maximum suppression

**Comment by phamquiluan** [comment]
https://pypi.org/project/memory-profiler/

This is a python module for monitoring memory consumption of a process as well as line-by-line analysis of memory consumption for python programs. It is a pure python module which depends on the psutil module.

**Comment by phamquiluan** [comment]
Accepted Manuscript
**Weighted Patch-based Manifold Regularization Dictionary Pair Learning
model for facial expression recognition using Iterative Optimization**
Classification Strategy
Link: https://sci-hub.tw/https://www.sciencedirect.com/science/article/pii/S1077314219300943

Á đù cái ni dính tới Regularization - Chính quy hóa =)))

Cách phân biệt high bias và high variance: bias liên tưởng tới sự lệch lạc, cái model với đám data lệch nhau quá nhiều, kiểu thằng model trong lúc học không care dữ liệu lắm (underfit). Còn variance thì vocabulary.com định nghĩa như sau **A change to the norm called variance**, cái variance ở đây có thể hiểu là sự đú theo các kiểu dữ liệu, điểm nào cũng muốn meet hết, -> overfitting.

to read: xgboost ??

**Comment by phamquiluan** [comment]
Accepted Manuscript
**Weighted Patch-based Manifold Regularization Dictionary Pair Learning
model for facial expression recognition using Iterative Optimization**
Classification Strategy
Link: https://sci-hub.tw/https://www.sciencedirect.com/science/article/pii/S1077314219300943

Á đù cái ni dính tới Regularization - Chính quy hóa =)))

Cách phân biệt high bias và high variance: bias liên tưởng tới sự lệch lạc, cái model với đám data lệch nhau quá nhiều, kiểu thằng model trong lúc học không care dữ liệu lắm (underfit). Còn variance thì vocabulary.com định nghĩa như sau **A change to the norm called variance**, cái variance ở đây có thể hiểu là sự đú theo các kiểu dữ liệu, điểm nào cũng muốn meet hết, -> overfitting.

to read: xgboost ??

**Comment by phamquiluan** [comment]
Đã bao lâu không đọc paper:
https://arxiv.org/pdf/1710.03740.pdf

MIXED PRECISION TRAINING

They cool abstract:
Increasing the size of a neural network typically improves accuracy but also increases the memory and compute requirements for training the model. We introduce methodology for training deep neural networks using half-precision floating point numbers, without losing model accuracy or having to modify hyperparameters. This nearly halves memory requirements and, on recent GPUs,
speeds up arithmetic. Weights, activations, and gradients are stored in IEEE halfprecision format. Since this format has a narrower range than single-precision we
propose three techniques for preventing the loss of critical information. Firstly,
we recommend maintaining a single-precision copy of weights that accumulates
the gradients after each optimizer step (this copy is rounded to half-precision for
the forward- and back-propagation). Secondly, we propose loss-scaling to preserve gradient values with small magnitudes. Thirdly, we use half-precision arithmetic that accumulates into single-precision outputs, which are converted to halfprecision before storing to memory. We demonstrate that the proposed methodology works across a wide variety of tasks and modern large scale (exceeding 100
million parameters) model architectures, trained on large datasets.

**Comment by phamquiluan** [comment]
https://towardsdatascience.com/review-ssd-single-shot-detector-object-detection-851a94607d11

**Comment by phamquiluan** [comment]
X11, strictly reading, a communication protocol.

http://www.wassen.net/x11-clipboard.html
https://unix.stackexchange.com/questions/276168/what-is-x11-exactly


GNOME is a free and open-source desktop environment for Unix-like operating systems. GNOME was originally an acronym for GNU Network Object Model Environment, but the acronym was dropped because it no longer reflected the vision of the GNOME project.

What is a desktop environment?
In computing, a desktop environment is an implementation of the desktop metaphor made of a bundle of programs running on top of a computer operating system, which share a common graphical user interface, sometimes described as a graphical shell.

Ôi dm lại đi quá xa rồi

**Comment by phamquiluan** [comment]
Like many words, “X11” can have multiple meanings.

“X11” is, strictly speaking, a communication protocol. In the sentences “X primarily defines protocol and graphics primitives …” and “X does not mandate the user interface …”, that's what X refers to. X is a family of protocols, X11 is the 11th version and the only one that's been in use in the last 25 years or so.

The first sentence in your question refers to a software distribution which is the reference implementation of the X11 protocol. The full name of this software distribution is “the X Window System”. This distribution includes programs that act as servers in the X11 protocol, programs that act as clients in the X11 protocol, code libraries that contain code that makes use of the X11 protocol, associated documentation, resources such as fonts and keyboard layouts that can be used with the aforementioned programs and libraries, etc. Historically, this software distribution was made by MIT; today it is maintained by the X.Org Foundation.

The X11 protocol allows applications to create objects such as windows and use basic drawing primitives (e.g. fill a rectangle, display some text). Widgets like buttons, menus, etc. are made by client libraries. The X Window System includes a basic library (the Athena widget set) but most applications use fancier libraries such as GTK+, Qt, Motif, etc.

Some X11 programs don't have a graphical user interface at all, for example command line tools such as xset, xsel and xdotool, key binding programs such as xbindkeys, etc. Most X11 programs do of course have a GUI.

**Comment by phamquiluan** [comment]
Note, that there are two distinct mechanisms to copy and paste in X11: the selection and the clipbard. The selection is used by highlighting some text with the mouse and pasted with the middle button (which is, on most mice, the wheel). To be precise, there are primary and secondary selections, but the latter is hardly used. The clipboard uses Ctrl-C and Ctrl-V to copy and paste (or the menu entries below Edit).

**Comment by phamquiluan** [comment]
má cái thằng này dịch vcl 

![image](https://user-images.githubusercontent.com/24642166/62307188-5a5e9280-b4ad-11e9-93a7-ed2a955e3075.png)

**Comment by phamquiluan** [comment]
Speed/accuracy trade-offs for modern convolutional object detectors

https://arxiv.org/pdf/1611.10012.pdf

**Comment by phamquiluan** [comment]
![image](https://user-images.githubusercontent.com/24642166/62315690-13c66380-b4c0-11e9-89f9-afd0f9f5cba8.png)

**Comment by phamquiluan** [comment]
daemon: background process

**Comment by phamquiluan** [comment]
docker cheat sheet: https://www.docker.com/sites/default/files/Docker_CheatSheet_08.09.2016_0.pdf

**Comment by phamquiluan** [comment]
tqdm means "progress" in Arabic (taqadum, تقدّم) and is an abbreviation for "I love you so much" in Spanish (te quiero demasiado).

https://github.com/tqdm/tqdm

**Comment by phamquiluan** [comment]
về bộ kiến trúc CNN cho bài toán detection, thường thì nó sẽ để một đám anchor/boxes/priors (tui gọi là anchor cho gọn) mặc định rồi tính toán 2 công thức sau:

F_loc = (I, a, theta) và F_cls(I, a, theta). Trong đó I là image, a là cái box, theta là bộ trọng số của model. Câu hỏi bây giờ là mỗi box sẽ được lưu trữ như thế nào? (có thể là x,y,w,h hay một cái gì đó tương tự được encoded và thể thể decode ra được x,y,w,h)

**Comment by phamquiluan** [comment]
https://en.wikipedia.org/wiki/Cerebral_cortex

đọc giải trí nào các bạn ;vv

**Comment by phamquiluan** [comment]
Gabor filter và một đống filter khác, một thanh niên chưa học môn xử lý ảnh cho hay
http://openlab.forumvi.com/t10-topic

https://www.google.com/search?rlz=1C1CHBF_enVN855VN855&ei=pINKXYyIKoy7wAPD7brgCA&q=x%E1%BB%AD+l%C3%BD+%E1%BA%A3nh+c%C3%A1c+b%E1%BB%99+l%E1%BB%8Dc&oq=x%E1%BB%AD+l%C3%BD+%E1%BA%A3nh+c%C3%A1c+&gs_l=psy-ab.1.0.0i22i30l10.8603.9259..11184...1.0..3.116.758.7j1......0....1..gws-wiz.......0.Uw9otB5OXk0

**Comment by phamquiluan** [comment]
lobe : thùy 

![image](https://user-images.githubusercontent.com/24642166/62607144-c9ae0980-b927-11e9-9eb2-ee39d181a117.png)

**Comment by phamquiluan** [comment]
quantum mechanics: cơ lượng tử

**Comment by phamquiluan** [comment]
vcl cái /etc bên linux đúng là etcetera thiệt @@
